component/core/src/main/java/org/exoplatform/social/core/profilelabel/ProfileLabelServiceImpl.java

```diff
@@ -0,0 +1,95 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.exoplatform.social.core.profilelabel;
+
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+
+import org.exoplatform.commons.ObjectAlreadyExistsException;
+import org.exoplatform.social.core.model.ProfileLabel;
+import org.exoplatform.social.core.profilelabel.storage.ProfileLabelStorage;
+
+public class ProfileLabelServiceImpl implements ProfileLabelService {
+
+  private final ProfileLabelStorage profileLabelStorage;
+
+  public ProfileLabelServiceImpl(ProfileLabelStorage profileLabelStorage) {
+    this.profileLabelStorage = profileLabelStorage;
+  }
+
+  @Override
+  public ProfileLabel findLabelByObjectTypeAndObjectIdAndLang(String objectType, String objectId, String language) {
+    return profileLabelStorage.findLabelByObjectTypeAndObjectIdAndLang(objectType, objectId, language);
+  }
+
+  @Override
+  public List<ProfileLabel> findLabelByObjectTypeAndObjectId(String objectType, String objectId) {
+    return profileLabelStorage.findLabelByObjectTypeAndObjectId(objectType, objectId);
+  }
+
+  @Override
+  public void createLabel(ProfileLabel profileLabel) throws ObjectAlreadyExistsException {
+    if (profileLabel == null) {
+      throw new IllegalArgumentException("ProfileLabel Item Object is mandatory");
+    }
+    if (StringUtils.isBlank(profileLabel.getObjectType())) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    if (profileLabel.getObjectId() == null) {
+      throw new IllegalArgumentException("Object Id is mandatory");
+    }
+    if (StringUtils.isBlank(profileLabel.getLanguage())) {
+      throw new IllegalArgumentException("language is mandatory");
+    }
+    if (StringUtils.isBlank(profileLabel.getLabel())) {
+      throw new IllegalArgumentException("profileLabel value is mandatory");
+    }
+    ProfileLabel storedProfileLabel = profileLabelStorage.findLabelByObjectTypeAndObjectIdAndLang(profileLabel.getObjectType(),
+                                                                                                  profileLabel.getObjectId(),
+                                                                                                  profileLabel.getLanguage());
+    if (storedProfileLabel != null) {
+      throw new ObjectAlreadyExistsException(storedProfileLabel, "ProfileLabel already exist");
+    }
+    profileLabel.setId(null);
+    profileLabelStorage.saveLabel(profileLabel, true);
+  }
+
+  @Override
+  public void updateLabel(ProfileLabel profileLabel) {
+    if (StringUtils.isBlank(profileLabel.getObjectType())) {
+      throw new IllegalArgumentException("Object type is mandatory");
+    }
+    if (profileLabel.getObjectId() == null) {
+      throw new IllegalArgumentException("Object Id is mandatory");
+    }
+    profileLabelStorage.saveLabel(profileLabel, false);
+  }
+
+  @Override
+  public void deleteLabel(Long id) {
```

Comment:
No need to use an object for long here, besides an NPE can be thrown

```suggestion
  public void deleteLabel(long id) {
```